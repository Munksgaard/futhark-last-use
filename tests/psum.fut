let psum = scan (+) 0

let main (xss: [][]i32) =
  #[incremental_flattening(only_intra)]
  map (psum >-> psum >-> psum)
      xss
