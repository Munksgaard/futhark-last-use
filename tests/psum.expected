Up to date
(res_419, [x_417, x_418])
(x_420, [xss_247, gtid_292, gtid_295])
(res_429, [x_427, x_428])
(x_430, [gtid_297, resarr0_416])
(res_438, [x_436, x_437])
(x_439, [gtid_299, resarr0_426])
(bytes_763, [binop_x_764])
(mem_765, [bytes_763])
(bytes_767, [binop_x_768])
(mem_769, [bytes_767])
(bytes_771, [binop_x_772])
(mem_773, [bytes_771])
(bytes_774, [binop_x_777])
(binop_x_777, [binop_x_775, binop_y_776])
(mem_778, [bytes_774])

